utils/ansible/ftest/templates/daos-make.sh.j2

```diff
@@ -196,21 +204,49 @@ if [[ ! -d "$DAOS_INSTALL_DIR" ]] ; then
 	fi
 fi
 
-cd "$DAOS_SOURCE_DIR"
-SCONS_OPTS="-j $JOBS_NB --implicit-deps-changed"
-case $CMD in
-	install) SCONS_OPTS="$SCONS_OPTS --build-deps=yes --config=force" ;;
-	update) SCONS_OPTS="$SCONS_OPTS --build-deps=no" ;;
-	*) fatal "Unexpected case" ;;
-esac
-
-info "Building and installing DAOS from source tree $PWD to $DAOS_INSTALL_DIR"
-if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE=debug PREFIX="$DAOS_INSTALL_DIR" install $SCONS_OPTS ; then
-	fatal "DAOS could not be properly build and/or installed"
+if [[ ! -d "$DAOS_VENV_DIR" ]] ; then
+	info "Creating DAOS python virtualenv directory $DAOS_VENV_DIR"
+	if ! run mkdir -p "$DAOS_VENV_DIR" ; then
+		fatal "DAOS python virtualenv directory $DAOS_VENV_DIR could not be created"
+	fi
+fi
+
+if [[ ! -f "$DAOS_VENV_DIR/bin/activate" ]] ; then
+	info "Creating DAOS python virtualenv $DAOS_VENV_DIR"
+	run $PYTHON_EXE -m venv "$DAOS_VENV_DIR"
+fi
+info "Installing DAOS python virtualenv $DAOS_VENV_DIR"
+source "$DAOS_VENV_DIR/bin/activate"
+run $PIP_EXE install --upgrade pip
+run $PIP_EXE install -r "$DAOS_SOURCE_DIR/requirements.txt"
+
+SCONS_OPTS="--directory="$DAOS_SOURCE_DIR" --jobs=$JOBS_NB"
+
+if "$DAOS_BUILD_DEPS" ; then
+	info "Building DAOS dependencies from source tree $DAOS_SOURCE_DIR"
+	if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE="$DAOS_BUILD_TYPE" PREFIX="$DAOS_INSTALL_DIR" $SCONS_OPTS --build-deps=only ; then
+		fatal "DAOS dependencies could not be properly build"
+	fi
+fi
+
+info "Building DAOS from source tree $DAOS_SOURCE_DIR"
+if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE="$DAOS_BUILD_TYPE" PREFIX="$DAOS_INSTALL_DIR" $SCONS_OPTS --build-deps=no ; then
+	fatal "DAOS could not be properly build"
+fi
+
+info "Installing DAOS to $DAOS_INSTALL_DIR"
+source "$DAOS_VENV_DIR/bin/activate"
+if ! run env MPI_PKG=any "$SCONS_EXE" BUILD_TYPE="$DAOS_BUILD_TYPE" PREFIX="$DAOS_INSTALL_DIR" install $SCONS_OPTS ; then
```

Comment:
The opts are used here but they'll be saved in daos.conf by the invocation above.
```suggestion
if ! run "$SCONS_EXE" install $SCONS_OPTS ; then
```

Reply:
- [ ] Remove useless opts

Reply:
> * Remove useless opts

Fixed with commit 97ec61dff